src/pytest_cov/engine.py

```diff
@@ -107,29 +127,26 @@ def summary(self, stream):
             skip_covered = isinstance(self.cov_report, dict) and 'skip-covered' in self.cov_report.values()
             if hasattr(coverage, 'version_info') and coverage.version_info[0] >= 4:
                 options.update({'skip_covered': skip_covered or None})
-            total = self.cov.report(**options)
+            _work_around_config_mutation(self.cov, "report", **options)
 
         # Produce annotated source code report if wanted.
         if 'annotate' in self.cov_report:
             annotate_dir = self.cov_report['annotate']
-            self.cov.annotate(ignore_errors=True, directory=annotate_dir)
-            # We need to call Coverage.report here, just to get the total
-            # Coverage.annotate don't return any total and we need it for --cov-fail-under.
-            total = self.cov.report(ignore_errors=True, file=StringIO())
+            _work_around_config_mutation(self.cov, "annotate", ignore_errors=True, directory=annotate_dir)
             if annotate_dir:
                 stream.write('Coverage annotated source written to dir %s\n' % annotate_dir)
             else:
                 stream.write('Coverage annotated source written next to source\n')
 
         # Produce html report if wanted.
         if 'html' in self.cov_report:
-            total = self.cov.html_report(ignore_errors=True, directory=self.cov_report['html'])
-            stream.write('Coverage HTML written to dir %s\n' % self.cov.config.html_dir)
+            v = _work_around_config_mutation(self.cov, "html_report", ignore_errors=True, directory=self.cov_report['html'])
+            stream.write('Coverage HTML written to dir %s\n' % v.config.html_dir)
```

Comment:
This sets v to the return from html_report, which is a float, then uses v as if it were cov.

Reply:
v is a _CovResult